src/main/cpp/test/TPImpeller.cpp

```diff
@@ -0,0 +1,33 @@
+///Created by the one, the only, Big Chief, AJ Gdowski
+///Nobody will ever be brave enough to steal my code, so don't even think about it
+#include <test/TPImpeller.h>
+#include <controllers/impeller/ImpellerStateMgr.h>
+using namespace frc;
+
+TPImpeller::TPImpeller()
+{
+    m_timer = new Timer;
+
+}
+
+void TPImpeller::periodic()
+{
+    auto time = m_timer -> Get();
+    if (time < 15)
+    {
+        m_impellerStateMgr -> SetCurrentState(ImpellerStateMgr::IMPELLER_STATE::ON, true);
```

Comment:
m_impellerStateMgr is uninitialized; you need to construct it in the constructor just like you created m_timer